os/arch/arm/src/amebad/amebad_timer_lowerhalf.c

```diff
@@ -0,0 +1,356 @@
+		return OK;
+	}
+
+	/* Return EBUSY to indicate that the timer was already running */
+	tmrdbg("Error!! Timer %d was already running\n", priv->obj.timer_id);
+	return -EBUSY;
+}
+
+/****************************************************************************
+ * Name: amebad_gpt_stop
+ *
+ * Description:
+ *   Stop the timer
+ *
+ * Input Parameters:
+ *   lower - A pointer the publicly visible representation of the "lower-half"
+ *           driver state structure.
+ *
+ * Returned Values:
+ *   Zero on success; a negated errno value on failure.
+ *
+ ****************************************************************************/
+static int amebad_gpt_stop(struct timer_lowerhalf_s *lower)
+{
+	struct amebad_gpt_lowerhalf_s *priv = (struct amebad_gpt_lowerhalf_s *)lower;
+	DEBUGASSERT(priv);
+
+	if (priv->started) {
+		gtimer_deinit(&priv->obj);
+		priv->started = false;
+
+		tmrvdbg("Timer %d is stopped\n", priv->obj.timer_id);
+		return OK;
+	}
+
+	/* Return ENODEV to indicate that the timer was not running */
+	tmrdbg("Error!! Timer %d is not running\n", priv->obj.timer_id);
+	return -ENODEV;
+}
+
+/****************************************************************************
+ * Name: amebad_gpt_settimeout
+ *
+ * Description:
+ *   Set a new timeout value (and reset the timer)
+ *
+ * Input Parameters:
+ *   lower   - A pointer the publicly visible representation of the
+ *             "lower-half" driver state structure.
+ *   timeout - The new timeout value in microseconds.
+ *
+ * Returned Values:
+ *   Zero on success; a negated errno value on failure.
+ *
+ ****************************************************************************/
+static int amebad_gpt_settimeout(struct timer_lowerhalf_s *lower, uint32_t timeout)
+{
+	struct amebad_gpt_lowerhalf_s *priv = (struct amebad_gpt_lowerhalf_s *)lower;
+	DEBUGASSERT(priv);
+
+	if (priv->started) {
+		return -EPERM;
+	}
+
+	gtimer_reload(&priv->obj, timeout);
+	priv->gpt_timeout = timeout;
+
+	return OK;
+}
+
+/****************************************************************************
+ * Name: amebad_gpt_setcallback
+ *
+ * Description:
+ *   Set the user provided timeout callback.
+ *
+ * Input Parameters:
+ *   lower    - A pointer the publicly visible representation of the
+ *              "lower-half" driver state structure.
+ *   callback - The new timer expiration function pointer. If this
+ *              function pointer is NULL, then the reset-on-expiration
+ *              behavior is restored,
+ *  arg       - Argument that will be provided in the callback
+ *
+ * Returned Values:
+ *   The previous timer expiration function pointer or NULL is there was
+ *   no previous function pointer.
+ *
+ ****************************************************************************/
+static void amebad_gpt_setcallback(struct timer_lowerhalf_s *lower, tccb_t callback, void *arg)
+{
+	struct amebad_gpt_lowerhalf_s *priv = (struct amebad_gpt_lowerhalf_s *)lower;
+	DEBUGASSERT(priv);
```

Comment:
Sorry but DEBUGASSERT is not enough.
DEBUGASSERT is working only with CONFIG_DEBUG enabled. When CONFIG_DEBUG is disabled and priv is null, the system could be crashed due to null dereference like `priv ->callback`. Please check that with if condition.

Reply:
Could you check previous comment?

Reply:
> Could you check previous comment?

Sure, I was following the coding style that other in use platforms are using. I will also add 'if (priv)' along with DEBUGASSERT for all the functions. Thanks!